### ISSUE TITLE
Support sparse grids

### ISSUE THREAD

**Issue by casellimarco** [ott-jax/ott#618]
This is halfway between a feature request and an indication of what tool fits best with my problem.

**Is your feature request related to a problem? Please describe.**
My input is in the grid format, but it's really sparse, i.e. most locations have zero weight. 

**Describe the solution you'd like**
I would like the solver to take advantage of the null weights, dropping those entries from the geometry.

**Describe alternatives you've considered**
I have considered to use PointCloud as geometry but then I struggle to reshape my data without having issues with `jax.jit` transform, since the array shape is no more static but depends on the data.

**Comment by marcocuturi** [comment]
@casellimarco apologies for letting your question slip through. This was peak Neurips preparations and i forgot to answer it! 

It seems you will need to go the `PointCloud` route no matter what. However, if you're worried about varying support size, something that could be the middle point you mention is the `segment_sinkhorn` util, which essentially pads pointclouds with points of 0 mass so that computations can be jitted.

Again, apologies for the super late answer.
